### PR TITLE
Ai cv update bugs

### PR DESCRIPTION
**Pull request by pavelpetrov47**

### CODE REVIEW

**Review by PetkoLyuts — APPROVED** [review]

File: AI_CV_React/frontend/src/cv/CvTemplate.jsx [hunk 1]
```diff
@@ -45,13 +61,33 @@ const CvTemplate = ({
       });
   };
 
-  const handleSave = (updatedSummary) => {
+  const handleSaveExperience = (field, updatedValue, index) => {
+    console.log("Updating experience:", index);
+    console.log("Field:", field);
```
Comment (PetkoLyuts) [inline AI_CV_React/frontend/src/cv/CvTemplate.jsx:66]: logs

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PersonServiceImpl.java [hunk 2]
```diff
@@ -62,12 +63,15 @@ public void deleteByEmail(String email) {
...
-        personRepository.save(editedPerson);
-        return true;
+    public boolean updateById(Person person) {
+        Optional<Person> editedPerson = personRepository.findById(person.getId());
+        if (editedPerson.isPresent()) {
+            editedPerson.get().setSummary(person.getSummary());
+            editedPerson.get().setExperience(person.getExperience());
+            personRepository.save(editedPerson.get());
+            return true;
+        }
```
Comment (mkanevv) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PersonServiceImpl.java:73]: Add error handling

**Review by PetkoLyuts — APPROVED** [review]

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PdfServiceImpl.java [hunk 3]
```diff
@@ -72,24 +73,25 @@ public List<Education> mapEducation(JsonNode education) {
...
+
+            Technology technology;
+            if (existingTechnologyOptional.isPresent()) {
+                technology = existingTechnologyOptional.get();
+            } else {
+                technology = new Technology();
+                technology.setName(technologyName);
+                technology = technologyService.saveTechnology(technology);
+            }
+            technologies.add(technology);
```
Comment (mkanevv) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PdfServiceImpl.java:91]: Error handling

**Review by mkanevv — CHANGES_REQUESTED** [review]

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PdfServiceImpl.java (hunk 3 above)
Comment (pavelpetrov47) [reply AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PdfServiceImpl.java]: i dont need error handling here

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PersonServiceImpl.java (hunk 2 above)
Comment (pavelpetrov47) [reply AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PersonServiceImpl.java]: fixed

**Review by mkanevv — CHANGES_REQUESTED** [review]
dont forget to format

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PdfServiceImpl.java [hunk 4]
```diff
@@ -72,25 +75,25 @@ public List<Education> mapEducation(JsonNode education) {
...
+    public List<Technology> mapTechnologies(JsonNode technologiesNode) {
+        List<Technology> technologies = new ArrayList<>();
+        for (JsonNode technologyNode : technologiesNode) {
+            String technologyName = technologyNode.asText();
+            Optional<Technology> existingTechnologyOptional = technologyService.findTechnology(technologyName);
+
+            Technology technology;
+            if(existingTechnologyOptional.isPresent()) {
+                technology = existingTechnologyOptional.get();
+            }else{
```
[the remark this answers is not in the document]
Comment (pavelpetrov47) [reply AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/impl/PdfServiceImpl.java]: all set

**Review by mkanevv — APPROVED** [review]

**Pull request merged**